net/chord/net.go

```diff
@@ -141,23 +141,24 @@ func (t *TCPTransport) getConn(host string) (*tcpOutConn, error) {
 	// Check if we have a conn cached
 	var out *tcpOutConn
 	t.poolLock.Lock()
+	defer t.poolLock.Unlock()
```

Comment:
@yilunzhang I consider as move t.poolLock.Unlock() behind net.DialTimeout will decrease performance...
getConn was invoked frequently, it is sensitive with performance.

Reply:
Agree and fixed.